lib/Raven/Processor/SanitizeCookiesProcessor.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Raven\Processor;
+
+use Raven\Event;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This processor sanitizes the cookies to ensure no sensitive information are
+ * sent to the server.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class SanitizeCookiesProcessor implements ProcessorInterface
+{
+    /**
+     * @var array The configuration options
+     */
+    private $options;
+
+    /**
+     * Class constructor.
+     *
+     * @param array $options An optional array of configuration options
+     */
+    public function __construct(array $options = [])
+    {
+        $resolver = new OptionsResolver();
+
+        $this->configureOptions($resolver);
+
+        $this->options = $resolver->resolve($options);
+
+        if (!empty($this->options['only']) && !empty($this->options['except'])) {
+            throw new InvalidOptionsException('You can configure only one of "only" and "except" options.');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Event $event)
+    {
+        $request = $event->getRequest();
+
+        if (isset($request['cookies'])) {
+            $cookiesToSanitize = array_keys($request['cookies']);
+
+            if (!empty($this->options['only'])) {
+                $cookiesToSanitize = $this->options['only'];
+            }
+
+            if (!empty($this->options['except'])) {
+                $cookiesToSanitize = array_diff($cookiesToSanitize, $this->options['except']);
+            }
+
+            foreach ($request['cookies'] as $name => $value) {
```

Comment:
Maybe if you use an `array_filter` this function will be shorter and more readable?

Reply:
`array_filter` would remove the cookies that gets sanitized, while instead I want to retain them and mask their values. There could be a case where the user wants to know that a certain cookie was sent but it contains sensitive data and so he wants to sanitize it